src/lib.rs

```diff
@@ -243,3 +246,9 @@ pub use crate::term_like::TermLike;
 
 #[cfg(feature = "rayon")]
 pub use crate::rayon::ParallelProgressIterator;
+
+pub trait StatefulFormatter: Display + Send {
```

Comment:
This should live near the bottom of the `style` module, please rename it to `ProgressTracker`.